x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSearchRequestTests.java

```diff
@@ -56,6 +59,13 @@ protected NamedXContentRegistry xContentRegistry() {
     @Override
     protected EqlSearchRequest createTestInstance() {
         try {
+            List<FieldAndFormat> randomFetchFields = new ArrayList<>();
+            for (int j = 0; j < randomIntBetween(0, 5); j++) {
```

Comment:
Still getting you a random number of iterations, but the exit condition is (likely) going to change with each iteration.

Reply:
Indeed. Nice catch!